modules/product/commerce_product.module

```diff
@@ -44,6 +44,17 @@ function commerce_product_entity_form_display_update(EntityFormDisplayInterface
 }
 
 /**
+ * Implements hook_ENTITY_TYPE_update().
+ */
+function commerce_product_entity_view_display_update(Drupal\Core\Entity\EntityInterface $entity) {
+  // The product view uses the variation view and needs to be cleared, which doesn't happen automatically
+  // because we're editing the variation, not the product
+  if ($entity->getConfigTarget() == 'commerce_product_variation.default.default') {
```

Comment:
Shouldn't we target any `commerce_product_variation.*.*`? As you can embed variation displays in multiple cases with different view modes.